extensions/omni_httpd/omni_httpd.c

```diff
@@ -166,6 +166,11 @@ void _Omni_init(const omni_handle *handle) {
 #ifdef _SC_NPROCESSORS_ONLN
   default_num_http_workers = sysconf(_SC_NPROCESSORS_ONLN);
 #endif
+  // considering the omni_httpd master worker and the "omni startup" worker used by omni TODO:
+  // revisit once omni_workers is set.
+  int threshold = max_worker_processes - 2;
```

Comment:
I think we can't reasonably predict how many workers we can expect as this is subject to all installed extensions and their whims. What do you think?

Reply:
Yes, correct. I believe the simplest way to solve this for now is to change the default http workers to 1. And then provide the `0` value as `auto`. Then if we emit warnings is reasonable to say that auto tried to give the best perf but it couldn't because there are no available workers.

That way at least we don't emit warnings in the default case.

What do you think?

Reply:
My primary concern than low number of workers result in poor performance, and 1 results in sequential processing of requests, which is... sad.

I wish there was a way to tell postgres that some workers can be easily scaled down under demand (other extensions wanting to register workers)

I'd suggest keeping what we have here and iterating further separately.

Reply:
My primary concern than low number of workers result in poor performance, and 1 results in sequential processing of requests, which is... sad.

I wish there was a way to tell postgres that some workers can be easily scaled down under demand (other extensions wanting to register workers)

I'd suggest keeping what we have here and iterating further separately.